go/multipooler/pools/admin/admin_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admin
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/multigres/multigres/go/common/fakepgserver"
+	"github.com/multigres/multigres/go/multipooler/pools/connpool"
+)
+
+func newTestPool(_ *testing.T, server *fakepgserver.Server) *Pool {
+	pool := NewPool(&PoolConfig{
+		ClientConfig: server.ClientConfig(),
+		ConnPoolConfig: &connpool.Config{
+			Capacity:     2,
+			MaxIdleCount: 2,
+		},
+	})
+	pool.Open(context.Background())
+	return pool
+}
+
+func TestPool_GetConnection(t *testing.T) {
+	server := fakepgserver.New(t)
+	defer server.Close()
+	server.SetNeverFail(true) // Allow any query
+
+	pool := newTestPool(t, server)
+	defer pool.Close()
+
+	ctx := context.Background()
+
+	// Get a connection.
+	pooled, err := pool.Get(ctx)
+	require.NoError(t, err)
+	require.NotNil(t, pooled)
+
+	// Verify connection is working.
+	assert.False(t, pooled.Conn.IsClosed())
+
+	// Return connection.
+	pooled.Recycle()
+
+	// Verify stats.
+	stats := pool.Stats()
+	assert.Equal(t, int64(1), stats.Active)
+	assert.Equal(t, int64(1), stats.Idle)
+}
+
+func TestPool_TerminateBackend(t *testing.T) {
+	server := fakepgserver.New(t)
+	defer server.Close()
+
+	// Set up expected pg_terminate_backend response.
+	server.AddQueryPattern(`SELECT pg_terminate_backend\(\d+\)`, fakepgserver.MakeResult(
+		[]string{"pg_terminate_backend"},
+		[][]any{{"t"}}, // "t" = true in PostgreSQL
+	))
+
+	pool := newTestPool(t, server)
+	defer pool.Close()
+
+	ctx := context.Background()
+
+	// Terminate a backend.
+	success, err := pool.TerminateBackend(ctx, 12345)
```

Comment:
I don't think the fake is verifying that pg_terminate_backend() was actually called, so it seems like this test would pass even if TerminateBackend() was a no-op?

Could/should the fake server add some kind of t.Cleanup() check to make sure all registered query patterns were actually used?